router/customdestinationmanager/customdestinationmanager_test.go

```diff
@@ -197,3 +269,154 @@ func TestKVManagerInvocations(t *testing.T) {
 		customManager.send(event, mockKVStoreManager, someDestination.Config)
 	})
 }
+
+func TestRedisManagerForJSONStorage(t *testing.T) {
+	initCustomerManager()
+	customManager := New("REDIS", Opts{}).(*CustomManagerT)
+	someDestination := backendconfig.DestinationT{
+		ID: "someDestinationID1",
+		DestinationDefinition: backendconfig.DestinationDefinitionT{
+			Name: "REDIS",
+		},
+	}
+	err := customManager.onNewDestination(someDestination)
+	assert.Nil(t, err)
+
+	for _, tc := range redisJSONTestCases {
+		t.Run(tc.description, func(t *testing.T) {
+			redisAddr, destroy := testutils.StartRedis(t, tc.redisImgRepo, tc.redisImgTag)
+			defer destroy()
+			event, err := json.Marshal(tc.transformedResponse)
+			if err != nil {
+				t.Errorf("MarshalError: %v\n", err.Error())
+				return
+			}
```

Comment:
Less code plus internally `t.Errorf` calls `Fail()` only.
```suggestion
			require.NoError(t, err)
```

Reply:
Probably for other instances as well.

Reply:
Done